src/test/java/core/basesyntax/service/impl/CalculatorImplTest.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.service.impl;
+
+import static core.basesyntax.model.Product.BANANA;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Calculator;
+import core.basesyntax.strategy.impl.OperationStrategyImpl;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class CalculatorImplTest {
+    private static Calculator calculator;
+
+    @BeforeAll
+    static void beforeAll() {
+        calculator = new CalculatorImpl(new OperationStrategyImpl());
+    }
+
+    @DisplayName("Check calculator with valid FruitTransaction")
+    @Test
+    public void calculate_validTransaction_ok() {
+        List<FruitTransaction> transactions =
+                List.of(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                        BANANA.getName(), 20));
+        calculator.calculate(transactions);
+        Map<String, Integer> expected = Map.of(BANANA.getName(), 20);
+        assertEquals(expected, Storage.STORAGE_MAP);
+    }
+
+    @DisplayName("Check calculator with invalid FruitTransaction")
+    @Test
+    public void calculate_invalidTransaction_notOk() {
+        List<FruitTransaction> transactions =
+                List.of(new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                        BANANA.getName(), 20));
+        calculator.calculate(transactions);
+        Map<String, Integer> expected = Map.of(BANANA.getName(), 30);
+        assertNotEquals(expected, Storage.STORAGE_MAP);
```

Comment:
I don't understand what the case this method is check. We put 20 bananas and check that 20 not equals to 30? 
If we wont to test some negative scenario we should e.g. pass some wrong parameter and check that exception was thrown.
CalculatorImp have the loop inside that calls different handlers. So CalculatorImpleTest can just test how the loop is works (several transations passed) the seperate transaction processing we will test in specific handler testing classes (like BalanceOperationHandlerTest etc)